alvr/client_core/src/platform/android.rs

```diff
@@ -393,12 +390,40 @@ pub fn video_decoder_split(
 
             image_reader
                 .set_image_listener(Box::new({
-                    let acquired_image = Arc::clone(&acquired_image);
-                    let image_acquired_notifier = Arc::clone(&image_acquired_notifier);
+                    let image_queue = Arc::clone(&image_queue);
                     move |image_reader| {
-                        let mut acquired_image_lock = acquired_image.lock();
-                        *acquired_image_lock = image_reader.acquire_next_image();
-                        image_acquired_notifier.notify_one();
+                        let mut image_queue_lock = image_queue.lock();
+
+                        if image_queue_lock.len() > available_buffering_frames {
+                            warn!("Video frame queue overflow!");
+                            image_queue_lock.clear();
```

Comment:
I think this is a bit extreme. Maybe just pop front?

Reply:
I think it's a good idea.